presto-main-base/src/main/java/com/facebook/presto/security/AccessControlManager.java

```diff
@@ -159,8 +164,21 @@ protected void setSystemAccessControl(String name, Map<String, String> propertie
         SystemAccessControlFactory systemAccessControlFactory = systemAccessControlFactories.get(name);
         checkState(systemAccessControlFactory != null, "Access control %s is not registered", name);
 
-        SystemAccessControl systemAccessControl = systemAccessControlFactory.create(ImmutableMap.copyOf(properties));
-        this.systemAccessControl.set(systemAccessControl);
+        if (Boolean.parseBoolean(properties.get(ACCESS_CONTROL_DETAILED_STATS))) {
+            log.info("Enabling detailed stats recording for system access control");
+
+            properties = new HashMap<>(properties);
+            properties.remove(ACCESS_CONTROL_DETAILED_STATS);
+            StatsRecordingSystemAccessControl statsRecordingSystemAccessControl =
+                    new StatsRecordingSystemAccessControl(systemAccessControlFactory.create(ImmutableMap.copyOf(properties)));
+            // Export the stats JMX MBean
+            exporter.export(generatedNameOf(AccessControlManager.class, "DetailedSystemAccessControlStats"), statsRecordingSystemAccessControl.getStats());
```

Comment:
Our convention is to export this from Guice, not by hand, e.g. `newExporter(binder).export(CacheStats.class).withGeneratedName()`

Reply:
The way you could do it is, instead of wiring in the `MBeanExporter` in `AccessControlManager`, you would wire in the Stats object and expose it as a singleton, e.g. 

```java
        binder.bind(CacheStats.class).in(Scopes.SINGLETON);
        newExporter(binder).export(CacheStats.class).withGeneratedName();
```

The advantage of doing it this way is it's not hidden in the access control class, it's at the top of the dependency chain where it's really obvious that `AccessControlManager` is playing some hand in recording stats.  I suggest this because the way I usually search for stats is to search for `newExporter(...`, it's a pretty reliable clue as to how to find stats classes.